src/allmydata/web/directory.py

```diff
@@ -82,33 +81,53 @@ def make_handler_for(node, client, parentnode=None, name=None):
         return DirectoryNodeHandler(client, node, parentnode, name)
     return UnknownNodeHandler(client, node, parentnode, name)
 
-class DirectoryNodeHandler(RenderMixin, rend.Page, ReplaceMeMixin):
-    addSlash = True
+
+class DirectoryNodeHandler(ReplaceMeMixin, Resource, object):
 
     def __init__(self, client, node, parentnode=None, name=None):
-        rend.Page.__init__(self)
+        super(DirectoryNodeHandler, self).__init__()
         self.client = client
         assert node
         self.node = node
         self.parentnode = parentnode
         self.name = name
         self._operations = client.get_web_service().get_operations()
 
-    def childFactory(self, ctx, name):
-        name = name.decode("utf-8")
+    def getChild(self, name, req):
+        """
+        Dynamically create a child for the given request and name
+        """
+        # trying to replicate what I have observed as Nevow behavior
+        # for these nodes, which is that a URI like
+        # "/uri/URI%3ADIR2%3Aj...vq/" (that is, with a trailing slash
+        # or no further children) renders "this" page
+        name = name.decode('utf8')
         if not name:
-            raise EmptyPathnameComponentError()
+            # replicating Nevow behavior that complains about "empty
+            # path segments" .. but twisted.web sends in "name=None"
+            # for a URL like "/foo/bar/" as well as "/foo//bar"
+            # (i.e. a trailing slash means "name=None" as well)
+            if b'//' in req.path:
+                raise EmptyPathnameComponentError(
+                    u"The webapi does not allow empty pathname components, i.e. a double slash",
+                )
+            return self
```

Comment:
https://github.com/tahoe-lafs/tahoe-lafs/blob/master/src/allmydata/test/web/test_web.py#L1747

Reply:
Of course, just because there's an existing test for it doesn't make it a great idea :/

Reply:
Okay.  Maybe I can convince myself at least that I know what the code is trying to accomplish.  And this PR isn't the place to question the behavior Tahoe-LAFS presents since the PR is just about porting.  So, great, carry on.